### PR TITLE
adds tracing for checks

### PR DESCRIPTION
**Pull request by christophkloeffel**
adds tracing for chapter 10

### CODE REVIEW

File: language-reference-manual/lrm.trlc [hunk 1]
```diff
@@ -1001,6 +1001,7 @@ section "Checks" {
           error "CB Version must be positive", version
       }
       """
+      untraced_reason = "Example code"
```
Comment (florianschanda) [inline language-reference-manual/lrm.trlc:1004]: Illegal - CI should have complained about this:
```
untraced_reason = "Example code"
^^^^^^^^^^^^^^^ language-reference-manual/lrm.trlc:1004: error: cannot overwrite frozen component untraced_reason
```

I have created https://github.com/bmw-software-engineering/trlc/issues/37 to fix that.

File: language-reference-manual/lrm.trlc [hunk 2]
```diff
@@ -886,7 +886,7 @@ section "Checks" {
               order.'''
   }
 
-  Dynamic_Semantics Check_Evaluation_Order_Across_Blocks {
+  Note Check_Evaluation_Order_Across_Blocks {
```
Comment (florianschanda) [inline language-reference-manual/lrm.trlc:889]: Good point. This is actually also a bug in the linter, and I have created https://github.com/bmw-software-engineering/trlc/issues/36 to fix it.

File: tests-system/checks-10/tracing [hunk 3]
```diff
@@ -0,0 +1 @@
+LRM.Applicable_Components
```
Comment (florianschanda) [inline tests-system/checks-10/tracing:1]: Like with PR #35 please use the `rbt-` naming scheme for new tests that test specific requirements.

**Review by florianschanda — CHANGES_REQUESTED** [review]
### LRM.Check_Block

Needs link to:
* `ast.Check_Block`
* `ast.Check`

### LRM.Check_Evaluation_Order

Needs link to:
* `ast.Record_Object.perform_checks`
* `ast.Composite_Type.add_check`
* `ast.Composite_Type.iter_checks`
* `ast.Record_Type.iter_checks`
* `ast.Tuple_Type.perform_type_checks`

### LRM.Check_Evaluation_Order_For_Extensions

I do not think the link to `parser.Parser.parse_check_block` makes a
lot of sense.

Needs link to:
* `ast.Record_Type.iter_checks`

**Comment by christophkloeffel** [comment]
> ### LRM.Check_Block
> Needs link to:
> 
> * `ast.Check_Block`
> * `ast.Check`
> 
After tracing a class it is not possible to trace the methods within anymore. They get ignored in the tracing report. Is this intended? 

as of now `LRM.Check_Messages` and `LRM.Check_Severity` are not traced to `ast.Check.perform`

**Pull request merged**